src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -643,6 +632,18 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("[]");
             return null;
           }
+          // Arguments<?> and NodeList<?> in es3 externs are correspondinly
+          // IArguments and NodeList interfaces (not-parametrized) in lib.d.ts.
+          // New* are temporary work-arounds for upstream externs.
+          // TODO(rado): upgrade closure compiler and remove them.
+          if (type.getDisplayName().equals("Arguments") || type.getDisplayName().equals("NewArguments")) {
+            emit("IArguments");
+            return null;
+          }
+          if (type.getDisplayName().equals("NodeList") || type.getDisplayName().equals("NewNodeList")) {
```

Comment:
Are you suggesting data driven config, or just code-wise extract it. Because I don't think we know the nature of the beast yet, to create a nice data DSL. These examples might be a bit deceiving. think about `Array.<?>` vs `any[]`.